### PR TITLE
Use LZ4-compressed system image in AOSP-EG APK

### PR DESCRIPTION
**Pull request by hlwarriner**
b/348014682

Change-Id: Id5039cba694ef992f28f9a7efe4192cd45c76803

### CODE REVIEW

**Comment by hlwarriner** [comment]
A small fix is also made for arm64, which was hard-coded to always copy the cobalt binary.

File: starboard/build/config/BUILDCONFIG.gn [hunk 1]
```diff
@@ -408,12 +408,21 @@ template("evergreen_loader") {
...
+    } else {
+      library_extension = "so"
+    }
+
     if (target_cpu == "arm" && arm_float_abi == "softfp") {
-      sources = [ "$root_out_dir/../evergreen-$target_cpu-${arm_float_abi}_$build_type/install/lib/lib${original_target_name}.so" ]
+      sources = [ "$root_out_dir/../evergreen-$target_cpu-${arm_float_abi}_$build_type/install/lib/lib${original_target_name}.${library_extension}" ]
     } else if (target_cpu == "arm64") {
-      sources = [ "$root_out_dir/../evergreen-$target_cpu_$build_type/install/lib/libcobalt.so" ]
+      sources = [ "$root_out_dir/../evergreen-$target_cpu_$build_type/install/lib/lib${original_target_name}.${library_extension}" ]
```
Comment (oxve) [inline starboard/build/config/BUILDCONFIG.gn:423]: Nit: I think this is slightly clearer. You choose how you want it.

```
    abi_suffix = ""
    if (target_cpu == "arm" && arm_float_abi == "softfp") {
      abi_suffix = "-${arm_float_abi}"
    }
    sources = [ "$root_out_dir/../evergreen-$target_cpu${abi_suffix}_$build_type/install/lib/lib${original_target_name}.${library_extension}" ]
```

**Review by oxve — APPROVED** [review]
Otherwise lgtm

File: starboard/build/config/BUILDCONFIG.gn (hunk 1 above)
Comment (hlwarriner) [reply starboard/build/config/BUILDCONFIG.gn]: Agreed, thanks.

**Review by johnxwork — APPROVED** [review]

**Review by yuying-y — APPROVED** [review]

**Pull request merged**